File: c++/web_tutorial_mastercopy/imdb_snippet.cpp

#include <iostream>
#include <string>
#include <vector>

using namespace std;

#include "Bridges.h"
#include "DataSource.h"
#include "data_src/ActorMovieIMDB.h"

using namespace bridges;

// This program fragment illustrates how to access and read the IMDB actor/movie data
int main(int argc, char **argv) {
		
	// create Bridges object
#if TESTING
    // command line args provide credentials and server to test on
    Bridges bridges (atoi(argv[1]), argv[2], argv[3]);
    if (argc > 4)
        bridges.setServer(argv[4]);
#else
    Bridges bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", 
                                "YOUR_API_KEY");
#endif

    // set title
	bridges.setTitle("Accessing IMDB Data");

	// create data source object
	DataSource ds (&bridges);

	// get the actor movie IMDB data through the BRIDGES API
	vector<ActorMovieIMDB> actor_list = ds.getActorMovieIMDBData(1814);

	// print out the first record of the dataset
	cout << "Actor-Movie Data:" << endl 
		<< "\tActor: " << actor_list[0].getActor() << endl 
		<< "\tMovie: " << actor_list[0].getMovie() << endl;

	return 0;
}

File: python/web_tutorial_mastercopy/imdb_snippet.py
from bridges.bridges import *
from bridges.data_src_dependent.data_source import *
import sys
import random

def main():

    # create the Bridges object, set credentials
#if TESTING
    args = sys.argv[1:]
    # command line args provide credentials and server to test on
    args = sys.argv[1:]
    bridges = Bridges(int(args[0]), args[1], args[2])
    if len(args) > 3:
        bridges.connector.set_server(args[3])
#else
    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY")
#endif

    # get imdb actor movie data
    my_list = get_actor_movie_imdb_data()

    # get an example pair of the data
    pair1 = my_list[random.randrange(len(my_list))]

    # print the actor and movie
    print(pair1.actor)
    print(pair1.movie)

if __name__ == "__main__":
    main()
